### DISCUSSION TITLE
Error 500 When Deploying Cattr App on GCP Debian 11 Instance

### DISCUSSION THREAD

**Discussion by shwetd19** [cattr-app/server-application#288] [Q&A]
Hey There,

I am trying to deploy the Cattr application on a fresh GCP Debian 11 instance, but I'm encountering a 500 error when visiting the URL. Here's a detailed account of the steps I followed and the issues I'm facing:

**Steps Followed:**

1. Spun up a Fresh Instance of Debian 11
2. Updated Repos and Upgraded Preinstalled Apps
```bash
sudo apt update && sudo apt upgrade -y
```

3. Installed Basic Prerequisites:
```bash
sudo apt install git wget sudo curl wget gnupg gnupg2 sb-release apt-transport-https ca-certificates software-properties-common -y
```

4. Installed MySQL 8.0:
```bash
wget https://dev.mysql.com/get/mysql-apt-config_0.8.30-1_all.deb
sudo dpkg -i mysql-apt-config_0.8.30-1_all.deb
sudo wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
sudo sh -c 'echo "deb https://packages.sury.org/php/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/php.list'
sudo apt update
sudo apt install mysql-server -y
sudo apt install php8.2 php8.2-{bcmath,bz2,intl,gd,mbstring,mysql,zip,curl,xml} -y
sudo apt install nginx php8.2-fpm -y
```

5. Uninstalled Apache:
Noticed that running sudo apt install nginx php8.2-fpm -y resulted in Apache2 getting installed, causing nginx to fail when I attempted to start it using systemctl restart nginx. So, I uninstalled Apache:
```bash
sudo systemctl stop apache2
sudo apt-get purge apache2 apache2-utils apache2-bin apache2.2-common
sudo apt-get remove apache2
sudo apt-get autoremove
sudo rm -rf /etc/apache2
sudo rm -rf /var/www
sudo rm -rf /var/log/apache2
sudo rm -rf /var/lib/apache2
```

6. Restarted nginx and php8.2-fpm:
```bash
sudo systemctl restart nginx
sudo systemctl restart php8.2-fpm
```

7. Installed Node.js v18:
```bash
curl -fsSL https://deb.nodesource.com/setup_18.x | sudo -E bash -
sudo apt install -y nodejs
```

8. Installed Yarn:
```bash
curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
sudo apt update
sudo apt install yarn
```

9. Installed Composer:
```bash
php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
php -r "if (hash_file('sha384', 'composer-setup.php') === 'dac665fdc30fdd8ec78b38b9800061b4150413ff2e3b6f88543c636f7cd84f6db9189d43a81e5503cda447da73c7e5b6') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
php composer-setup.php
php -r "unlink('composer-setup.php');"
sudo mv composer.phar /usr/local/bin/composer
```

10. Cloned the Repo:
```bash
cd /opt
git clone https://git.amazingcat.net/cattr/core/app.git
cd app
```
11. Started Installation:
```bash
sudo composer install && sudo php artisan key:generate && sudo yarn
sudo php artisan migrate --seed --seeder=InitialSeeder
sudo php artisan cattr:make:admin
npm run prod
```

**Issues Encountered:**

- When visiting the URL at http://10.10.10.218, I get a 500 Error with almost no information to go on.
- Checked the storage/logs/laravel.log, only saw an older error message about Mix manifest not found at: /opt/app/public/mix-manifest.json which has been resolved.
- Chrome browser console shows WebSocket connection errors:
WebSocket connection to wss://10.10.10.218/app/cattr?protocol=7&client=js&version=8.4.8-rc2&flash=false failed.

**Additional Information:**
- Followed the installation manual from the documentation website:
```bash
composer install
php artisan key:generate
yarn
```
- Edited .env file for DB connection using .env.example as reference
```bash
php artisan migrate --seed --seeder=InitialSeeder
```
- App won't start without seeding InitialSeeder.
```bash
php artisan cattr:make:admin
```

- Login with: Email: admin@cattr.app    &    Password: password

- Run Local server
```bash
php artisan serve
yarn dev
```
Generate IDE helpers:
```bash
composer dumphelpers
composer dumperd
```

Can anyone suggest how to troubleshoot this 500 error? Any insights or additional steps I should follow would be greatly appreciated!

Thanks in advance!

**Comment by saumiktewari** [comment]
@vermorag : hey!! i followed the steps above and am facing the same Error 500 issue as well. Love your help here please

**Comment by saumiktewari** [comment]
@amazing-cat guys any help will be appreciated
